src/features/vote/ActiveRequests.tsx

```diff
@@ -33,7 +64,43 @@ const ActiveRequests: FC<Props> = ({ activeRequests }) => {
         </div>
       </div>
       {activeRequests.length ? (
-        <ActiveRequestsForm activeRequests={activeRequests} />
+        <ActiveRequestsForm
+          publicKey={publicKey}
+          isConnected={isConnected}
+          activeRequests={activeRequests}
+          votePhase={votePhase}
+          encryptedVotes={encryptedVotes}
+          refetchEncryptedVotes={refetchEncryptedVotes}
+        />
+      ) : null}
+      {activeRequests.length &&
+      votePhase === "Reveal" &&
+      round.snapshotId === "0" ? (
+        <Button
+          onClick={() => {
+            if (!signer || !votingContract || !provider) return;
+            votingContract.functions["snapshotMessageHash"]().then((hash) => {
+              const sigHash = hash[0];
+              if ((window as any).ethereum) {
+                const mm = (window as any).ethereum;
+                const Web3 = new web3(mm);
+                if (address) {
+                  getMessageSignatureMetamask(Web3, sigHash, address).then(
```

Comment:
in the spirit of getting things working this is fine, but this is a good candidate for a refactor when we have more time